core/impl/query_index_manager.cxx

```diff
@@ -135,7 +117,7 @@ class watch_context : public std::enable_shared_from_this<watch_context>
         return timeout_ - std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - start_time_);
     }
 
-    bool check(const couchbase::core::operations::management::query_index_get_all_response& resp)
+    bool check(couchbase::core::operations::management::query_index_get_all_response resp)
```

Comment:
Why is this not a const reference anymore?

Reply:
Because finish() takes the resp as a reference, and we are modifying its error code there

Reply:
Could we make this a non-const reference instead, to avoid making a copy of the response when `check()` is called?